iscservice/demo/util.c

```diff
@@ -0,0 +1,710 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ */
+
+#include <stdio.h>
+#include <unistd.h>           /* gethostname */
+
+#include "lib/semaphore.h"    /* m0_semaphore */
+#include "lib/trace.h"        /* m0_trace_set_mmapped_buffer */
+#include "rpc/rpclib.h"       /* m0_rpc_post_sync */
+#include "motr/client.h"
+#include "motr/client_internal.h" /* m0_reqh */
+#include "conf/helpers.h"     /* m0_confc_root_open */
+#include "conf/diter.h"       /* m0_conf_diter_next_sync */
+#include "conf/obj_ops.h"     /* M0_CONF_DIRNEXT */
+#include "spiel/spiel.h"      /* m0_spiel_process_lib_load */
+#include "reqh/reqh.h"        /* m0_reqh */
+#include "layout/plan.h"      /* m0_layout_io_plop */
+
+#include "iscservice/isc.h"
+#include "util.h"
+
+#ifndef DEBUG
+#define DEBUG 0
+#endif
+
+enum {
+	MAX_M0_BUFSZ = 128*1024*1024, /* max bs for object store I/O  */
+	MAX_POOLS = 16,
+	MAX_RCFILE_NAME_LEN = 512,
+	MAX_CONF_STR_LEN = 128,
+	MAX_CONF_PARAMS = 32,
+};
+
+/* static variables */
+static struct m0_client          *m0_instance = NULL;
+static struct m0_container container;
+static struct m0_config    m0_conf;
+static struct m0_idx_dix_config   dix_conf;
+static struct m0_spiel            spiel_inst;
+
+static char c0rcfile[MAX_RCFILE_NAME_LEN] = "./.m0utilrc";
+
+/*
+ ******************************************************************************
+ * GLOBAL VARIABLES
+ ******************************************************************************
+ */
+struct m0_realm uber_realm;
+unsigned unit_size = 0;
+int trace_level=0;
+bool m0trace_on = false;
+
+struct param {
+	char name[MAX_CONF_STR_LEN];
+	char value[MAX_CONF_STR_LEN];
+};
+
+static struct m0_fid pools[MAX_POOLS] = {};
```

Comment:
This is very specific to SAGE cluster.
Remove it.
As a demo app, we don't need this.
A fid of pool is enough.

Reply:
Agree, removed.